pkg/sql/codegen/proto/intermediate_representation.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
```

Comment:
The proto file needs more comments like `TrainIR` since it will be read by many developers.

Reply:
I'll merge this first and do these polishments in another PR.